### PR TITLE
 feat(brigade.js): build docker images on tag event.

### PR DESCRIPTION
**Pull request by technosophos**
This adds a new target that builds and releases to a Docker registry.

Wait to merge until after #409

### CODE REVIEW

File: brigade.js [hunk 1]
```diff
@@ -127,10 +127,171 @@ function ghNotify(state, msg, e, project) {
...
+  // Docker builder
+  const dind = new Job("dind", "docker:stable-dind")
+  dind.storage.enabled = true
+  dind.privileged = true
+  dind.env = {
+    DOCKER_DRIVER: "overlay"
+  }
+  dind.tasks = [
+    "dockerd-entrypoint.sh &",
+    "echo waiting for Docker && sleep 20",
```
Comment (mumoshu) [inline brigade.js]: For no flakiness, `printf "waiting for docker daemon"; while ! docker info >/dev/null 2>&1; do printf .; sleep 1; done; echo` could be used instead, if you like.

Comment (technosophos) [reply brigade.js]: :hugs: This works great!

File: brigade.js [hunk 2]
```diff
@@ -127,10 +128,169 @@ function ghNotify(state, msg, e, project) {
...
+
+  return Group.runEach([goBuild, dind]);
+}
+
+function acrBuild(e, project, tag) {
+  const gopath = "/go"
+  const localPath = gopath + "/src/github.com/" + project.repo.name;
+  const org = "deis";
+  const images = [
+    //"brig", // Uncomment this after 0.13.0.
```
Comment (adamreese) [inline brigade.js]: 0.13 is shipped

File: brigade.js [hunk 3]
```diff
@@ -127,10 +128,169 @@ function ghNotify(state, msg, e, project) {
...
+
+  // For each image we want to build, build it, then tag it latest, then post it to registry.
+  for (let i of images) {
+    let imgName = registryHost+"/"+i+":"+tag;
+    let latest = registryHost+"/"+i+":latest";
+    dind.tasks.push(
+      `cd ${i}`,
+      `echo '========> Building ${i}'`,
+      `cp -a /mnt/brigade/share/${i}/rootfs ./rootfs`,
+      // Total hack to work around something weird with brigade-github-gateway
```
Comment (adamreese) [inline brigade.js]: what does this fix?

File: brigade.js [hunk 4]
```diff
@@ -127,10 +128,169 @@ function ghNotify(state, msg, e, project) {
...
+      `[ ! -f /mnt/brigade/share/${i}/rootfs/${i} ] || cp /mnt/brigade/share/${i}/rootfs/${i} ./rootfs/`,
+      // TODO: Fix the Makefile to make this easier.
```
Comment (adamreese) [inline brigade.js]: `make docker-build docker-push IMAGE_TAG=latest`

Comment (technosophos) [reply brigade.js]: I'll update the comment on that. But the problem with `make docker-....` is that I always have to rebuild from source, which means always having to load the Go toolchain. Here, I am copying the binary from another location so I don't have to recompile.

File: brigade.js (hunk 3 above)
Comment (technosophos) [reply brigade.js]: It fixes this peculiar case where the brigade-github-gateway bin is not copied by the `cp` above. I haven't been able to trace it down. There are no errors... it's just skipped.

File: brigade.js (hunk 2 above)
Comment (technosophos) [reply brigade.js]: I'll fix

File: brigade.js [hunk 5]
```diff
@@ -127,10 +128,169 @@ function ghNotify(state, msg, e, project) {
...
+  for (let i of images) {
+    let imgName = org +"/"+i+":"+tag;
+    let latest = org +"/"+i+":latest";
+    builder.tasks.push(
+      `cd ${i}`,
+      `echo '========> Building ${i}'`,
+      `cp -a /mnt/brigade/share/${i}/rootfs ./rootfs`,
+      // Total hack to work around something weird with brigade-github-gateway
+      `[ ! -f /mnt/brigade/share/${i}/rootfs/${i} ] || cp /mnt/brigade/share/${i}/rootfs/${i} ./rootfs/`,
+      // Currently, it appears this is the only way to add two tags.
```
Comment (ehotinger) [inline brigade.js]: 💄 You can remove this comment since it's not applicable anymore. I forgot to delete it from my previous PR.

**Pull request merged**